packages/nextjs/components/blockexplorer/TransactionsTable.tsx

```diff
@@ -0,0 +1,139 @@
+import Link from "next/link";
+import { TransactionResponse } from "@ethersproject/abstract-provider";
+import { ethers } from "ethers";
+import { Address } from "~~/components/scaffold-eth";
+
+type TransactionWithFunction = TransactionResponse & { functionName?: string };
+
+// Define your types
+interface Block {
+  hash: string;
+  transactions: TransactionWithFunction[];
+  timestamp: number;
+  number: number;
+}
+
+interface ConfiguredNetwork {
+  nativeCurrency: {
+    symbol: string;
+  };
+}
+
+interface TransactionReceipts {
+  [key: string]: {
+    contractAddress: string;
+  };
+}
+
+interface TransactionsTableProps {
+  blocks: Block[];
+  transactionReceipts: TransactionReceipts;
+  configuredNetwork: ConfiguredNetwork;
+}
+
+export const TransactionsTable: React.FC<TransactionsTableProps> = ({
+  blocks,
+  transactionReceipts,
+  configuredNetwork,
+}) => {
+  return (
+    <div className="overflow-x-auto shadow-lg">
+      <table className="min-w-full divide-y divide-primary shadow-lg rounded-lg bg-neutral">
+        <thead className="bg-primary">
+          <tr>
+            <th
+              scope="col"
+              className="px-6 py-3 text-left text-xs font-medium text-primary-content uppercase tracking-wider"
+            >
+              Transaction Hash
+            </th>
+            <th
+              scope="col"
+              className="px-6 py-3 text-left text-xs font-medium text-primary-content uppercase tracking-wider"
+            >
+              Function Called
+            </th>
+            <th
+              scope="col"
+              className="px-6 py-3 text-left text-xs font-medium text-primary-content uppercase tracking-wider"
+            >
+              Block Number
+            </th>
+            <th
+              scope="col"
+              className="px-6 py-3 text-left text-xs font-medium text-primary-content uppercase tracking-wider"
+            >
+              Time Mined
+            </th>
+            <th
+              scope="col"
+              className="px-6 py-3 text-left text-xs font-medium text-primary-content uppercase tracking-wider"
+            >
+              From
+            </th>
+            <th
+              scope="col"
+              className="px-6 py-3 text-left text-xs font-medium text-primary-content uppercase tracking-wider"
+            >
+              To
+            </th>
+            <th
+              scope="col"
+              className="px-6 py-3 text-left text-xs font-medium text-primary-content uppercase tracking-wider"
+            >
+              Value ({configuredNetwork.nativeCurrency.symbol})
+            </th>
+          </tr>
+        </thead>
+        <tbody className="bg-base-100 divide-y divide-primary-content text-base-content">
+          {Array.from(blocks.reduce((map, block) => map.set(block.hash, block), new Map()).values()).map(block =>
```

Comment:
Doubt : 
Why can't we directly use `blocks.map()` 😅 ?

Reason : 
```ts
  const blocks = [Block1, Block2, Block2]

  const blockObj = blocks.reduce(map, block) => map.set(block.hash, block), new Map<string, Block>())
  // blockObj = {
  // "0x1234": Block1,
  // "0x5678": Block2,
  // "0x9012": Block3,
  // }

  const values = blockObj.values()
  // values = MapIterator {Block1, Block2, Block3}

  const finalArr = Array.from(values)
  // finalArr = [Block1, Block2, Block3]
```
therefore finalArr == blocks, why don't we use `blocks.map` directly 😅


I am sure I might have missed something here, was just trying to keep intuitive and understand , but we can keep it as is